components/tikv_util/src/codec/stream_event.rs

```diff
@@ -1,13 +1,56 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use crate::Either;
 use bytes::{Buf, Bytes};
 use std::io::prelude::*;
 use std::io::Cursor;
-use tikv_util::Either;
 
-pub struct Encoder;
+pub struct EventIterator {
+    buf: Cursor<Vec<u8>>,
+    index: usize,
+    len: usize,
+    pub val: Vec<u8>,
+}
+
+impl EventIterator {
+    pub fn new(buf: Vec<u8>) -> EventIterator {
+        let len = buf.len();
+        EventIterator {
+            buf: Cursor::new(buf),
+            index: 0,
+            len,
+            val: vec![],
+        }
+    }
+}
+
+impl Iterator for EventIterator {
+    type Item = Vec<u8>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.index >= self.len {
+            None
+        } else {
+            let len = self.buf.get_u32_le() as usize;
+            self.index += 4;
+            let mut key = vec![0; len];
+            self.buf.read_exact(key.as_mut_slice()).unwrap();
+            self.index += len;
+
+            let len = self.buf.get_u32_le() as usize;
+            self.index += 4;
+            let mut val = vec![0; len];
```

Comment:
we can maintain an internal buffer here for reducing allocation then. (Maybe not for now?)

Reply:
but if we reduce the allocation， we still need one more `clone` for this buffer. 🤔，so read buffer is unnecessary?